pocketflow/__init__.py

```
@@ -0,0 +1,65 @@
+        if isinstance(action,str): return _ConditionalTransition(self,action)
+        raise TypeError("Action must be a string")
+
+class _ConditionalTransition:
+    def __init__(self,src,action): self.src,self.action=src,action
+    def __rshift__(self,tgt): return self.src.add_successor(tgt,self.action)
+
+class Node(BaseNode):
+    def __init__(self,max_retries=1,wait=0): super().__init__();self.max_retries,self.wait=max_retries,wait
+    async def exec_fallback(self,prep_res,exc): raise exc
+    async def _exec(self,prep_res):
+        for self.cur_retry in range(self.max_retries):
```

Ruff (0.8.2)

Loop control variable `self` overrides iterable it iterates

(B020)